### PR TITLE
Add TableauConnectedApp model and associated Django Admin form

### PR DESCRIPTION
**Pull request by AddisonDunn**
## Product Description

This PR covers the first two tickets ([USH-2504](https://dimagi-dev.atlassian.net/browse/USH-2504?atlOrigin=eyJpIjoiNmVmZjk0YjY0YzMxNGRhYjhlM2MyYjYxNmFhMzMwMDIiLCJwIjoiaiJ9) and [USH-2505](https://dimagi-dev.atlassian.net/browse/USH-2505?atlOrigin=eyJpIjoiMmI5ZWJjNjdmMDVmNDcyMzljN2E5MThiNTY3MDFiNDAiLCJwIjoiaiJ9)) of the Tableau user syncing feature.

If you're curious, you can find the design for this feature [here](https://docs.google.com/document/d/1sPKHSNZWEMddjLYdBXvUSXViSbV0jgWw4b7MiB3PDPY/edit#bookmark=id.pi7o593ta46v).

## Technical Summary

The TableauConnectedApp model stores information that will allow HQ to authenticate with the Tableau API in order to use it for adding/syncing users between HQ and Tableau. The secret value is encrypted using [ideas taken from the Motech project](https://github.com/dimagi/commcare-hq/blob/6b3085c80662b3aa94cfcf3b73c4cb4080278786/corehq/motech/models.py#L103) that Ethan once recocomended.

The PyJWT dependency is added with this PR in order to create a JSON Web Token for authentication with the Tableau API. As Simon noted, this dependency [is already required](https://github.com/dimagi/commcare-hq/blob/980f5d82e3e9229371d76ccb521affa5f3f72afc/requirements/prod-requirements.txt#L379) via pygithub and twilio, this PR declares just declares it explicitly.

This is what the Admin form looks like: 
Screen Shot 2022-10-20 at 12 27 35 PM

## Feature Flag

The tableau user syncing feature will eventually be specific to the EMBEDDED_TABLEAU feature flag and a new feature flag for tableau user syncing.

### Safety story

Tested locally. This just adds new models and should have no end-user impact.

### Automated test coverage

Tests added. Domain deletion tests should determine whether the TableauConnectedApp model is deleted properly upon a domain being deleted.

### QA Plan

Will be done at the end of this feature's work.

### Migrations

- [X] The migrations in this code can be safely applied first independently of the code

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: requirements/base-requirements.in [hunk 1]
```diff
@@ -81,6 +81,7 @@ python-dateutil
 python-imap
 pyphonetics
 python-magic
+pytjwt
```
Comment (millerdev) [inline requirements/base-requirements.in:84]: Should this be `pyjwt`?

Also, I think it is necessary to run `make requirements` to update dependency comments in the requirements files even though this is already implicitly required.

File: corehq/apps/reports/models.py [hunk 2]
```diff
@@ -1,17 +1,19 @@
-from datetime import datetime
+import jwt
+import uuid
+
+from datetime import datetime, timedelta
 
 from django.conf import settings
-from django.core.exceptions import ValidationError
```
[the remark this answers is not in the document]
Comment (gherceg) [reply corehq/apps/reports/models.py:4]: For future reference, removing unused imports is great, but doing so in a separate commit dedicated to removing unused imports is even better. That way you don't have to make a comment like this 😁

File: requirements/base-requirements.in (hunk 1 above)
Comment (millerdev) [reply requirements/base-requirements.in]: :+1: looks good

**Review by orangejenny — APPROVED** [review]
Looks good, my comments are optional.

File: corehq/apps/reports/admin.py [hunk 3]
```diff
@@ -1,7 +1,35 @@
...
+
+# The encrypted field on the TableauConnectedApp model requires a custom admin form field.
+class TableauConnectedAppAdminForm(forms.ModelForm):
+    secret_value = forms.CharField(required=False,
+                                   help_text="Field may be empty even if value has been provided.",
+                                   widget=forms.PasswordInput)
+
+    def save(self, commit=True):
+        unencrypted_secret_value = self.cleaned_data.get('secret_value', None)
+        instance = super(TableauConnectedAppAdminForm, self).save(commit=commit)
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/reports/admin.py:19]: ah, I get it now, thanks for explaining

**Review by orangejenny — APPROVED** [review]

**Pull request merged**